python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -213,3 +221,128 @@ def pairwise_linestring_distance(offsets1, xs1, ys1, offsets2, xs2, ys2):
             )
         }
     )
+
+
+def pairwise_point_linestring_distance(
+    points: GeoSeries, linestrings: GeoSeries
+):
+    """Compute shortest distance between pairs of points and linestrings
+
+    The shortest distance between a pair of (multi)point and (multi)linestring
+    is defined as the shortest distance between every geometry in the
+    multipoint and every geometry in the multilinestring.
+
+    Parameters
+    ----------
+    points : GeoSeries
+        The points to compute the distance from.
+    linestrings : GeoSeries
+        The linestrings to compute the distance to.
```

Comment:
Be careful, this is not a directed distance. Also, "multi".

```suggestion
    points : GeoSeries
        The (multi)points to compute the distance from.
    linestrings : GeoSeries
        The (multi)linestrings to compute the distance from.
```